### PR TITLE
7 ne create reservation model

### LINKED ISSUE #7: Create Reservation model
Validate associations: 
belongs_to traveler (user)

attributes: 
status (maybe enum? confirmed, pending, canceled?)

### LINKED ISSUE #36: Create Night model
Associations:
belongs_to reservation_id (optional)
belongs_to couch_id

Attributes:
date

### PR DESCRIPTION
**Pull request by ski-climb**
closes #7 
closes #36 

Reservation model and Night model.  

@tmikeschu @njgheorghita @riverswb @Sh1pley Ready for review!

### CODE REVIEW

File: app/models/couch.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class Couch < ApplicationRecord
   validates_presence_of :name, :street_address, :city, :state, :zipcode
-  belongs_to :user
+  belongs_to :host, class_name: "User", foreign_key: :user_id
```
Comment (tmikeschu) [inline app/models/couch.rb:3]: Ruby is for reading! 👍

File: app/models/reservation.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Reservation < ApplicationRecord
+  validates_presence_of :status
+  belongs_to :traveler, class_name: "User", foreign_key: :user_id
+
+  enum status: %w(pending confirmed canceled)
```
Comment (njgheorghita) [inline app/models/reservation.rb:5]: cancelled?

File: db/migrate/20170208222643_set_default_status_for_reservations_to_pending.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class SetDefaultStatusForReservationsToPending < ActiveRecord::Migration[5.0]
+  def change
+    change_column :reservations, :status, :integer, default: 0
```
Comment (tmikeschu) [inline db/migrate/20170208222643_set_default_status_for_reservations_to_pending.rb:3]: Nice default 👌

File: spec/models/couch_spec.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@
   end
 
   context "relationships" do
-    it { should belong_to :user }
+    it { should belong_to(:host).class_name("User") }
```
Comment (tmikeschu) [inline spec/models/couch_spec.rb:13]: Good catch on this.

File: spec/models/night_spec.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+describe Night do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:date) }
+  end
+
+  describe "relationships" do
+    it { is_expected.to belong_to(:reservation) }
+    it { is_expected.to belong_to(:couch) }
```
Comment (tmikeschu) [inline spec/models/night_spec.rb:10]: Is there a reason to go with `is_expected.to` over `should`?

File: spec/models/reservation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  describe "#status" do
+    let!(:reservation) { create(:reservation) }
+
+    it "defaults to pending" do
+      expect(reservation.status).to eq "pending"
+      expect(reservation.pending?).to be true
+    end
+
+    it "can be set to confirmed" do
+      reservation.confirmed!
```
Comment (tmikeschu) [inline spec/models/reservation_spec.rb:21]: Enums!

File: spec/models/reservation_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    it { is_expected.to validate_presence_of(:status) }
+  end
+
+  describe "relationships" do
+    it { is_expected.to belong_to(:traveler).class_name("User") }
+  end
+
+  describe "#status" do
+    let!(:reservation) { create(:reservation) }
+
```
Comment (njgheorghita) [inline spec/models/reservation_spec.rb:14]: this is cool!

File: spec/models/reservation_spec.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:status) }
+  end
+
+  describe "relationships" do
+    it { is_expected.to belong_to(:traveler).class_name("User") }
+  end
+
+  describe "#status" do
+    let!(:reservation) { create(:reservation) }
```
Comment (tmikeschu) [inline spec/models/reservation_spec.rb:13]: Nice and dry. Thanks for explaining this!

**Review by tmikeschu — COMMENTED** [review]
Nice work, Mr. Nick! 👏

**Comment by njgheorghita** [comment]
nice work everything looks great to me, merge on!

**Pull request merged**